### PR TITLE
ETS-506 | Adding new fields

### PR DESCRIPTION
**Pull request by fwpushan**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -171,4 +171,19 @@ local-db-seed:
 
 local-server-tests:
 	@echo "Running tests in local app container"
-	@docker exec -it $(PROJECT_NAME)-server npm test
\ No newline at end of file
+	@docker exec -it $(PROJECT_NAME)-server npm test
+
+
+mongo-tunnel:
```
Comment (dbayly-freshworks) [inline makefile:177]: Nit - We might not want to commit these changes

**Review by dbayly-freshworks — COMMENTED** [review]
This looks ok to me. We should have this target testing instead of master though.

File: makefile (hunk 1 above)
Comment (fwkendall) [reply makefile:177]: Agreed, our bastion host was intended to be temporary anyway. We should actually coordinate with Vyas to get our readme updated with better instructions for the VPN connection, I think.

File: server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js [hunk 2]
```diff
@@ -1,23 +1,57 @@
...
-          { 'serviceBCTransactions.status': { $ne: 'success' } },
-          { parsed_arrival_date: { $gte: dateRange[0], $lte: dateRange[1] } },
+          {
+            "serviceBCTransactions.status": { $ne: "success" },
+            $and: [
+              {
+                $or: [
+                  {
+                    $and: [
+                      { prioritized_traveler: true },
```
Comment (fwkendall) [inline server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js:29]: I think the value comes through as the string "Yes" - does this work?

File: server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js [hunk 3]
```diff
@@ -27,32 +61,56 @@ const getUnsuccessfulSbcTransactions = async (collection, arrivalKey) => {
...
   return { id: submission.id, status: transaction.status };
 };
 
-const cleanJoinArray = (a) => a
-  .filter((i) => ['string', 'number'].includes(typeof i))
-  .map((i) => String(i).trim())
-  .filter((i) => i !== '')
-  .join(', ');
+const cleanJoinArray = (a) =>
+  a
```
Comment (fwkendall) [inline server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js:80]: do we want this style change?

File: server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js [hunk 4]
```diff
@@ -1,23 +1,57 @@
...
+const { postServiceItem } = require("./service-bc-api");
 
 const getUnsuccessfulSbcTransactions = async (collection, arrivalKey) => {
-  const dateRange = [dayjs().subtract(13, 'day'), dayjs().subtract(3, 'day')]
-    .map((d) => d.startOf('day').toDate());
+  const dateRange = [
+    dayjs().subtract(13, "day"),
+    dayjs().subtract(3, "day"),
+  ].map((d) => d.startOf("day").toDate());
+  // serviceBCTransactions.status != success && ((prioritized_traveler == true && parsed_arrival_date in range [d1, today]) || parsed_arrival_date in [d1, d2])
```
Comment (fwkendall) [inline server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js:10]: should remove this line

**Comment by fwkendall** [comment]
Looks good! But I think this needs a second pass just to clean up a couple of things we don't want to commit and confirm the query change works as intended for the boolean Yes/No value we're provided.

Just a note for future ideal PRs I guess as well, if we're going to be getting style updates to these older files we should attempt to make that its own commit so I can ignore it on PR and just focus on functional changes 😅

**Comment by fwkendall** [comment]
Also just a note this should be targeting test and not master :)

File: server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js (hunk 3 above)
Comment (fwkendall) [reply server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js:80]: David and I chatted - we think yes, just keep this for consistency with the prettier settings. Probably a good cue to us that in general we want to use more descriptive variable names even for very simple functions, though! I wouldn't worry about changing this as part of this PR

File: makefile (hunk 1 above)
Comment (fwpushan) [reply makefile:177]: I keep but hide all infra related info.

File: server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js (hunk 3 above)
Comment (fwpushan) [reply server/lambda/layer/common/nodejs/custom_modules/send-to-sbc.js:86]: Problem is that no prettier config file is available in repo. So, my VS code formatted with my default prettier setting.

**Review by dbayly-freshworks — APPROVED** [review]
Looks ok to me. It looks like Vyas committed some changes directly to prod, which has created a diff. I think the changes are harmless.

**Review by fwkendall — APPROVED** [review]

**Comment by fwkendall** [comment]
Yep looks good. This has the changes needed for the ticket + some additional changes due to the branches being out of sync. I can log a separate ticket for later to make sure we get that all cleaned up (and get our dev branch working again).

**Pull request merged**